arrow-meta/build.gradle

```
@@ -10,28 +10,28 @@ apply from: "$SUB_PROJECT"
 apply from: "$DOC_CREATION"
 
 dependencies {
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$KOTLIN_VERSION"
-    compile project(':arrow-annotations')
-    compile "me.eugeniomarletti.kotlin.metadata:kotlin-metadata:$KOTLIN_METADATA_VERSION"
-    compile("com.squareup:kotlinpoet:$KOTLIN_POET_VERSION") {
+    api "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$KOTLIN_VERSION"
```

I'm not sure this should be exposed as it imposes a specific version + jdk to other places, for example in our own repo we have jdk7, jdk8 and maybe some plain jdk declarations that would be duplicated

```suggestion
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$KOTLIN_VERSION"
```